rclcpp/src/rclcpp/duration.cpp

```diff
@@ -222,6 +245,67 @@ Duration::operator*(double scale) const
       static_cast<long double>(rcl_duration_.nanoseconds) * scale_ld));
 }
 
+Duration &
+Duration::operator*=(double scale)
+{
+  *this = *this * scale;
+  return *this;
+}
+
+Duration
+Duration::operator*(const rclcpp::Duration & rhs) const
+{
+  bounds_check_mul(
+    this->rcl_duration_.nanoseconds,
+    rhs.rcl_duration_.nanoseconds);
+  return Duration::from_nanoseconds(
+    rcl_duration_.nanoseconds * rhs.rcl_duration_.nanoseconds);
+}
+
+Duration &
+Duration::operator*=(const rclcpp::Duration & rhs)
+{
+  *this = *this * rhs;
+  return *this;
+}
+
+Duration
+Duration::operator/(double scale) const
+{
+  if (!std::isfinite(scale)) {
+    throw std::runtime_error("abnormal scale in rclcpp::Duration");
+  }
+  bounds_check_duration_scale(
+    this->rcl_duration_.nanoseconds,
+    1.0 / scale,
```

Comment:
scale could be zero, that leads to zero division runtime exception?

Reply:
This is a good catch, I removed division from this PR so it can be separate.